### PR TITLE
keyboards/ryanskidmore: add rskeys100 keyboard

### PR DESCRIPTION
**Pull request by ryanskidmore**
## Description

This PR adds the firmware for the `rskeys100` keyboard. This is a custom, full size, 100% ISO keyboard that I've built. This firmware uses custom matrix logic as the matrix PCB uses three shift registers to pulse over the columns.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ryanskidmore/rskeys100/matrix.c [hunk 1]
```diff
@@ -0,0 +1,176 @@
+/*
```
Comment (drashna) [inline keyboards/ryanskidmore/rskeys100/matrix.c:1]: Please convert this to the lite custom matrix. 

https://docs.qmk.fm/#/custom_matrix?id=lite

You can use the ergodox EZ or moonlander as an example of how to do this.

File: keyboards/ryanskidmore/rskeys100/rskeys100.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+         4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+         4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+         4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4
+     } };
+#endif
+
+void keyboard_post_init_user(void) {
+    rgb_matrix_enable_noeeprom();
+    rgb_matrix_mode_noeeprom(RGB_MATRIX_CYCLE_LEFT_RIGHT);
+}
```
Comment (drashna) [inline keyboards/ryanskidmore/rskeys100/rskeys100.c:35]: If you want to set a default mode, you can do so by by adding this to the config.h file. 
```c
#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_LEFT_RIGHT // Sets the default mode, if none has been set
```

Also, user function should not be in the keyboard's c file. 

```suggestion
```
or
```suggestion
void keyboard_post_init_kb(void) {
    rgb_matrix_enable_noeeprom();
    rgb_matrix_mode_noeeprom(RGB_MATRIX_CYCLE_LEFT_RIGHT);
    keyboard_post_init_user();
}
```

File: keyboards/ryanskidmore/rskeys100/rules.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no        # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/ryanskidmore/rskeys100/rules.mk:17]:
```suggestion
```

File: keyboards/ryanskidmore/rskeys100/rules.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no         # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+AUDIO_ENABLE = no            # Audio output
+CUSTOM_MATRIX = yes
```
Comment (drashna) [inline keyboards/ryanskidmore/rskeys100/rules.mk:24]:
```suggestion
CUSTOM_MATRIX = lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ryanskidmore/rskeys100/matrix.c (hunk 1 above)
Comment (ryanskidmore) [reply keyboards/ryanskidmore/rskeys100/matrix.c:1]: I've migrated to the `lite` implementation now - thanks for the heads up! The code looks a lot cleaner with a bunch of the boilerplate removed.

File: keyboards/ryanskidmore/rskeys100/rskeys100.c (hunk 2 above)
Comment (ryanskidmore) [reply keyboards/ryanskidmore/rskeys100/rskeys100.c]: I've removed the `rgb_matrix_mode_noeeprom` call and moved `keyboard_post_init_user` to `keymap.c`. Is this sufficient or should I revert to using `keyboard_post_init_kb` in the keyboard C file instead?

File: keyboards/ryanskidmore/rskeys100/rules.mk (hunk 3 above)
Comment (ryanskidmore) [reply keyboards/ryanskidmore/rskeys100/rules.mk]: Removed

File: keyboards/ryanskidmore/rskeys100/rules.mk (hunk 4 above)
Comment (ryanskidmore) [reply keyboards/ryanskidmore/rskeys100/rules.mk]: Migrated to the lite implementation, see above

File: keyboards/ryanskidmore/rskeys100/rskeys100.c (hunk 2 above)
Comment (drashna) [reply keyboards/ryanskidmore/rskeys100/rskeys100.c]: putting it in the keymap.c wound be fine, yeah.

Comment (ryanskidmore) [reply keyboards/ryanskidmore/rskeys100/rskeys100.c]: Sweet - ready for review in that case

**Comment by ryanskidmore** [comment]
All requested changes made @drashna - let me know if you have any other feedback 🙌

File: keyboards/ryanskidmore/rskeys100/config.h [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+ * the diode direction is COL(umn) to ROW */
+#define DIODE_DIRECTION COL2ROW
+
+/* RGB Data Pin */
+#define RGB_DI_PIN C7
+/* The number of RGB LEDs connected */
+#define DRIVER_LED_TOTAL 115
+/* Undefine, and then redefine the maximum brightness. This is set to 110 to avoid going over 500 mA.
+ * At full brightness with all three indicator LEDs on, the power draw is about 450 mA. */
+#undef RGB_MATRIX_MAXIMUM_BRIGHTNESS
```
Comment (drashna) [inline keyboards/ryanskidmore/rskeys100/config.h:44]: This shouldn't need to be undefined. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ryanskidmore/rskeys100/rules.mk [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no        # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/ryanskidmore/rskeys100/rules.mk:15]:
```suggestion
NKRO_ENABLE = yes            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by ryanskidmore** [comment]
Thanks both, all changes made.

**Review by drashna — APPROVED** [review]

File: keyboards/ryanskidmore/rskeys100/matrix.c [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+#include "util.h"
+#include "led.h"
+#include "config.h"
+#include "matrix.h"
+#include "quantum.h"
+#include "debounce.h"
+
+#ifndef DEBOUNCE
+#    define DEBOUNCE 5
+#endif
```
Comment (fauxpark) [inline keyboards/ryanskidmore/rskeys100/matrix.c:35]: Some of these may not be needed.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ryanskidmore/rskeys100/matrix.c (hunk 7 above)
Comment (ryanskidmore) [reply keyboards/ryanskidmore/rskeys100/matrix.c]: Good spot - I've removed the unused ones now.

File: keyboards/ryanskidmore/rskeys100/info.json [hunk 8]
```diff
@@ -0,0 +1,121 @@
+{
+    "keyboard_name": "rskeys100",
+    "url": "",
+    "maintainer": "ryanskidmore",
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/ryanskidmore/rskeys100/info.json:6]:
```suggestion
        "LAYOUT_fullsize_iso": {
```

File: keyboards/ryanskidmore/rskeys100/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+     * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┐ Ent│ └───┴───┴───┘ ├───┼───┼───┤ + │
+     * │ Caps │ A │ S │ D │ F │ G │ H │ J │ K │ L │ ; │ ' │ # │    │               │ 4 │ 5 │ 6 │   │
+     * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴────┤     ┌───┐     ├───┼───┼───┼───┤
+     * │Shft│ \ │ Z │ X │ C │ V │ B │ N │ M │ , │ . │ / │    Shift │     │ ↑ │     │ 1 │ 2 │ 3 │   │
+     * ├────┼───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤ ┌───┼───┼───┐ ├───┴───┼───┤Ent│
+     * │Ctrl│GUI │Alt │                        │ Alt│ GUI│Menu│Ctrl│ │ ← │ ↓ │ → │ │   0   │ . │   │
+     * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘ └───┴───┴───┘ └───────┴───┴───┘
+     */
+    /* Default Layer */
+    [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ryanskidmore/rskeys100/keymaps/default/keymap.c:24]:
```suggestion
    [0] = LAYOUT_fullsize_iso(
```

File: keyboards/ryanskidmore/rskeys100/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,    KC_INS,  KC_HOME, KC_PGUP,    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,     KC_DEL,  KC_END,  KC_PGDN,    KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS,                                           KC_P4,   KC_P5,   KC_P6,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,             KC_UP,               KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_APP,  KC_RCTL,    KC_LEFT, KC_DOWN, KC_RGHT,    KC_P0,            KC_PDOT
+    ),
+    /* RGB Control Layer */
+    [1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ryanskidmore/rskeys100/keymaps/default/keymap.c:34]:
```suggestion
    [1] = LAYOUT_fullsize_iso(
```

File: keyboards/ryanskidmore/rskeys100/rskeys100.h [hunk 11]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 Ryan Skidmore (@ryanskidmore, rskeys@ryanskidmore.co.uk)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/ryanskidmore/rskeys100/rskeys100.h:8]:
```suggestion
#define LAYOUT_fullsize_iso( \
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by ryanskidmore** [comment]
@noroadsleft Thank you for the feedback - all actioned!

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Pull request merged**